packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/StudioProjectTemplateInitializer.java

```diff
@@ -90,4 +117,12 @@ private Optional<ResourceSet> getResourceSet(IEditingContext editingContext) {
                 .map(IEMFEditingContext::getDomain)
                 .map(AdapterFactoryEditingDomain::getResourceSet);
     }
+
+    private Optional<DiagramDescription> findDomainDiagramDescription(IEditingContext editingContext) {
+        return this.representationDescriptionSearchService.findAll(editingContext).values().stream()
+                .filter(DiagramDescription.class::isInstance)
+                .map(DiagramDescription.class::cast)
+                .filter(diagramDescription -> Objects.equals("Domain", diagramDescription.getLabel()))
```

Comment:
This seems brittle to check on the label with a rather generic name; could we match on it's id and use a some random UUID to make sure we find the one we want?

Reply:
Given the recent improvements made in our tests, I think I now have what's needed for that. In the previous architecture it was only tested this way. I've updated the code.